hicommonwealth/commonwealth
# Contest rollover fix

## Link to Issue
Closes: #8057 
Closes: #8058
Closes: #8142

## Description of Changes
- Fixes bug where gas was too low to perform payload/rollover at end of contest
- Fixes bug where contest doesn't rollover due to txn not happening after contest has ended
- Fixes bug where voteContent onchain call was passing wrong content ID

## "How We Fixed It"
- Allow content add/vote to happen *after* a contest has ended, which is what triggers rollover

## Test Plan
- Create contest with `FLAG_CONTEST_DEV=true` so duration is 5 mins
- Create thread with staked vote
  - Ensure scores/winners show correctly
- Wait 5 mins until outside of contest window
- Create new thread
  - Ensure that new `Contest` DB entry is created

## Deployment Plan
N/A

## Other Considerations
N/A
